sourcemash/api/feeds.py

```diff
@@ -83,24 +83,39 @@ def get(self):
 
 class FeedAPI(Resource):
 
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('subscribed', type = inputs.boolean, default = True)
+        super(FeedAPI, self).__init__()
+
+
     def get(self, id):
         feed = Feed.query.get_or_404(id)
         return {'feed': marshal(feed, feed_fields)}
 
 
     @login_required
-    def delete(self, id):
-        """Unsubscribe user from feed."""
+    def put(self, id):
+        args = self.reqparse.parse_args()
 
-        try:
-            subscription = current_user.subscribed.filter(Feed.id==id).one()
-        except:
-            abort(404)
+        feed = Feed.query.get_or_404(id)
 
-        current_user.subscribed.remove(subscription)
-        db.session.commit()
+        if args.subscribed:
```

Comment:
This block is either executing/db-committing or erroring on every single put request. Is that the right behavior? I think so, but just wanted to make sure.